lightning/src/chain/onchaintx.rs

```diff
@@ -922,7 +966,12 @@ impl<ChannelSigner: WriteableEcdsaChannelSigner> OnchainTxHandler<ChannelSigner>
 					#[cfg(anchors)]
 					OnchainClaim::Event(claim_event) => {
 						log_info!(logger, "Yielding onchain event after reorg to spend inputs {:?}", request.outpoints());
-						self.pending_claim_events.insert(_first_claim_txid_height.0, claim_event);
+						if let Some((existing_claim_idx, _)) = self.pending_claim_events.iter().enumerate()
```

Comment:
same here. What's the benefit of a) stopping on the first find and b) not using retain?